src/biogeochem/CNBalanceCheckMod.F90

```diff
@@ -550,87 +656,98 @@ subroutine NBalanceCheck(this, bounds, num_soilc, filter_soilc, &
          write(iulog,*)'input mass               = ',col_ninputs(c)*dt
          write(iulog,*)'output mass              = ',col_noutputs(c)*dt
          write(iulog,*)'net flux                 = ',(col_ninputs(c)-col_noutputs(c))*dt
-         write(iulog,*)'inputs,ffix,nfix,ndep    = ',ffix_to_sminn(c)*dt,nfix_to_sminn(c)*dt,ndep_to_sminn(c)*dt
-         write(iulog,*)'outputs,lch,roff,dnit    = ',smin_no3_leached(c)*dt, smin_no3_runoff(c)*dt,f_n2o_nit(c)*dt
-         call endrun(subgrid_index=c, subgrid_level=subgrid_level_column, msg=errMsg(sourcefile, __LINE__))
-      end if
-
-      ! Repeat error check at the gridcell level
-      call c2g( bounds = bounds, &
-         carr = totcoln(bounds%begc:bounds%endc), &
-         garr = totgrcn(bounds%begg:bounds%endg), &
-         c2l_scale_type = 'unity', &
-         l2g_scale_type = 'unity')
-      call c2g( bounds = bounds, &
-         carr = col_ninputs_partial(bounds%begc:bounds%endc), &
-         garr = grc_ninputs_partial(bounds%begg:bounds%endg), &
-         c2l_scale_type = 'unity', &
-         l2g_scale_type = 'unity')
-      call c2g( bounds = bounds, &
-         carr = col_noutputs_partial(bounds%begc:bounds%endc), &
-         garr = grc_noutputs_partial(bounds%begg:bounds%endg), &
-         c2l_scale_type = 'unity', &
-         l2g_scale_type = 'unity')
-
-      err_found = .false.
-      do g = bounds%begg, bounds%endg
-         ! calculate the total gridcell-level nitrogen storage, for mass conservation check
-         ! Notes:
-         ! Not including seedn_grc in grc_begnb and grc_endnb because
-         ! seedn_grc forms out of thin air, for now, and equals
-         ! -1 * (dwt_seedn_to_leaf_grc(g) + dwt_seedn_to_deadstem_grc(g))
-         ! We account for the latter fluxes as inputs below; the same
-         ! fluxes have entered the pools earlier in the timestep. For true
-         ! conservation we would need to add a flux out of nfix into seed.
-         grc_endnb(g) = totgrcn(g) + tot_woodprod_grc(g) + cropprod1_grc(g)
-
-         ! calculate total gridcell-level inputs
-         grc_ninputs(g) = grc_ninputs_partial(g) + &
-                          dwt_seedn_to_leaf_grc(g) + &
-                          dwt_seedn_to_deadstem_grc(g)
-
-         ! calculate total gridcell-level outputs
-         grc_noutputs(g) = grc_noutputs_partial(g) + &
-                           dwt_conv_nflux_grc(g) + &
-                           product_loss_grc(g)
-
-         ! calculate the total gridcell-level nitrogen balance error for this time step
-         grc_errnb(g) = (grc_ninputs(g) - grc_noutputs(g)) * dt - &
-                        (grc_endnb(g) - grc_begnb(g))
-
-         if (abs(grc_errnb(g)) > this%nerror) then
-            err_found = .true.
-            err_index = g
+         if(col%is_fates(c))then
+            write(iulog,*)'inputs,ndep,nfix,suppn= ',ndep_to_sminn(c)*dt,nfix_to_sminn(c)*dt,supplement_to_sminn(c)*dt
```

Comment:
I think its there, no?

Reply:
OK, I think what I meant to say is that supplment_to_sminn should be included based on if supnitro state, rather than if fates is on or not. Also eventually we want FATES to have ndep for advanced modes of fates_pareh_mode, so that should be given based on that rather than if FATES or not. Let's discuss this a bit later.

Reply:
Actually it's maybe OK to always write out supplement_to_sminn, since it will be zero, when suplnitro=='NONE'. Unfortunately, I don't see something similar for ndep.

Reply:
#2068